includes/gateways/paypal/admin/connect.php

```diff
@@ -0,0 +1,478 @@
+		 * we're only actually getting the `user_id` back, which isn't super helpful. So although this
+		 * is coded to *potentially* support showing the email address of the account, in most (if not
+		 * all) cases it will just say "yes you're connected".
+		 *
+		 * @link https://developer.paypal.com/docs/api/identity/v1/#userinfo
+		 */
+
+		/* Translators: %s - the connected mode, either `sandbox` or `live` */
+		$mode_string    = edd_is_test_mode() ? __( 'sandbox', 'easy-digital-downloads' ) : __( 'live', 'easy-digital-downloads' );
+		$account_status = sprintf( __( 'Your PayPal account is successfully connected in %s mode.', 'easy-digital-downloads' ), $mode_string );
+
+		if ( ! empty( $response->emails ) && is_array( $response->emails ) ) {
+			foreach ( $response->emails as $email ) {
+				if ( ! empty( $email->value ) && ! empty( $email->primary ) ) {
+					$account_status = sprintf(
+					/* Translators: %1$s - PayPal account email; %2$s - the connected mode, either `sandbox` or `live` */
+						__( 'You are successfully connected to the account <strong>%1$s</strong> in %2$s mode.', 'easy-digital-downloads' ),
+						esc_html( $email->value ),
+						$mode_string
+					);
+				}
+			}
+		}
+
+		if ( ! edd_is_gateway_active( 'paypal_commerce' ) ) {
+			$account_status .= ' ' . sprintf(
+				/* Translators: %1$s opening anchor tag; %2$s closing anchor tag */
+					__( 'To start using PayPal, be sure to %1$senable it%2$s in the general gateway settings.', 'easy-digital-downloads' ),
+					'<a href="' . esc_url( admin_url( 'edit.php?post_type=download&page=edd-settings&tab=gateways&section=main' ) ) . '">',
+					'</a>'
+				);
+		}
+
+		/**
+		 * Now check the webhook connection.
+		 */
+		$webhook_status = wpautop( esc_html__( 'Webhook successfully configured for the following events:', 'easy-digital-downloads' ) );
+		$actions        = array(
+			'<button type="button" class="button edd-paypal-connect-action" data-nonce="' . esc_attr( wp_create_nonce( 'edd_update_paypal_webhook' ) ) . '" data-action="edd_paypal_commerce_update_webhook">' . esc_html__( 'Sync Webhook', 'easy-digital-downloads' ) . '</button>'
+		);
+		try {
+			$webhook = PayPal\Webhooks\get_webhook_details( $mode );
+			if ( empty( $webhook->id ) ) {
+				throw new \Exception();
+			}
+
+			// Now compare the events to make sure we have them all.
+			$expected_events = array_keys( PayPal\Webhooks\get_webhook_events( $mode ) );
+			$actual_events   = array();
+
+			if ( ! empty( $webhook->event_types ) && is_array( $webhook->event_types ) ) {
+				foreach ( $webhook->event_types as $event_type ) {
+					if ( ! empty( $event_type->name ) && ! empty( $event_type->status ) && 'ENABLED' === strtoupper( $event_type->status ) ) {
+						$actual_events[] = $event_type->name;
+					}
+				}
+			}
+
+			$missing_events = array_diff( $expected_events, $actual_events );
+			$number_missing = count( $missing_events );
+			if ( $number_missing ) {
```

Comment:
I think `array_diff` always returns an array, even if it's empty. Should there be a check to ensure that `count > 0`?

Reply:
`if ( $number_missing )` handles that. It won't make it into the `if` statement if `$number_missing` is `0`. :)